components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/signup/UserSelfRegistrationManager.java

```diff
@@ -738,10 +741,13 @@ private UserRecoveryData validateSelfRegistrationCode(String code, String verifi
                     verifiedChannelClaim);
         }
 
+        HashMap<String, String> userClaims = new HashMap<>();
         // Get the claims that needs to be updated.
         // NOTE: Verification channel is stored in Remaining_Sets in user recovery data.
-        HashMap<String, String> userClaims = getClaimsListToUpdate(user, recoveryData.getRemainingSetIds(),
-                externallyVerifiedClaim, recoveryData.getRecoveryScenario().toString());
+        if (!skipAccountConfirmation) {
```

Comment:
We have to skip  claim updates in introspection API right? Not just claim get. 
       updateUserClaims(userStoreManager, user, userClaims);

Reply:
Introduced a new private method just to retrieve the code details which avoids updating user claims or side effects related to code confirmation.